docs/proposals/logdir.md

```diff
@@ -0,0 +1,163 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/logdir.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# LogDir
+
+## Abstract
+
+A proposal for implementing `LogDir`. A `LogDir` can preserve and manage a pod's log files.
+
+Several existing issues and PRs were already created regarding that particular subject:
+* Collect logfiles inside the container [#12567](https://github.com/kubernetes/kubernetes/issues/12567)
+* Add LogDir [#13010](https://github.com/kubernetes/kubernetes/pull/13010#issuecomment-133997188)
+
+Author: WuLonghui (@wulonghui)
+
+## Motivation
+
+Some applications will write log to files, so need to add  `LogDir`. It can add some useful infomation to make logging agent easy to collect the logs, also we can manage the log files(like log rotation and maximum log file sizes).
```

Comment:
As someone who has hacked about 500 different ways trying to get apache to do this, I can see a lot of motivation here - definitely a frequent request. As part of supporting legacy apps, I'd like to pick something like this up.